### ISSUE TITLE
It seems like there is some kind of memory leak in consuming api

### ISSUE THREAD

**Issue by Flig** [djsavic1988/rabbitmq-cxx#1]
Your lib's api is fantastic, but even with your example -- `advconsumer`  -- all consumed messages didn't free memory, but i can't understand where...

**Comment by Flig** [comment]
i found out that i had to manually call `amqp_maybe_release_buffers_on_channel`

**Comment by djsavic1988** [comment]
Hi @Flig , sorry I couldn't get to this before, but I ran the advconsumer with valgrind using [rabbitmq-c v0.11.0](https://github.com/alanxz/rabbitmq-c/tree/v0.11.0) with ~100k messages and didn't get any leaks. Can you please post which version of rabbitmq-c are you using so I could potentially reproduce and patch this issue?

EDIT: No need to post anything, just noticed what that function (`amqp_maybe_release_buffers_on_channel`) is for (https://github.com/alanxz/rabbitmq-c/issues/211), I guess it should be called in the Channel destructor, will try to add it some time in the near future

**Comment by djsavic1988** [comment]
Upon reading more about this it seems that I was completely wrong, (2 times! :D), from what I've managed to find so far is that a call to maybe_release_buffers is required at some point. Need to investigate how to patch this (when and where to call, certainly not only for destruction)

**Comment by Flig** [comment]
It is enough for my project to call memory release after i acked all prefetched messages

**Comment by djsavic1988** [comment]
Handling of this should now be automatic.

I feared that this might affect the performance so I measured it and got funny results: The average Time was in the %2 range (gained or loss) depending on the run, while the CPU time was in the -7% (loss of performance) to +4% (gain) range. I've concluded that the change to automate releasing the buffers has not introduced any significant performance issues but can help with the memory usage of long running applications.
